AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -560,6 +560,28 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)__setNeedsLayout
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (!_flags.isMeasured) {
+    return;
+  }
+  
+  ASSizeRange oldConstrainedSize = _constrainedSize;
+  [self invalidateCalculatedLayout];
+  
+  if (_supernode) {
+    // Cause supernode's layout to be invalidated
+    [_supernode setNeedsLayout];
+  } else {
+    // This is the root node. Trigger a full measurement pass on *current* thread. Old constrained size is re-used.
+    [self measureWithSizeRange:oldConstrainedSize];
+    self.frame = CGRectMake(0.0f, 0.0f, _layout.size.width, _layout.size.height);
```

Comment:
I'm pretty sure this frame setter needs to preserve the origin.  This should work fine in table views, but imagine someone embedding a node in a different view hierarchy.  This would make it jump to the upper left.

Additionally, if there is a transform being applied, setting frame will be undefined.  So let's change this to a bounds setter.

Even the bounds setter needs to preserve the origin, in case it is a scrollable content area.  So if you change this to first read the bounds and only change the size, and set the bounds, I think we'll be good.